core/unit_test/CMakeLists.txt

```diff
@@ -284,6 +284,26 @@ TRIBITS_ADD_EXECUTABLE_AND_TEST(
     TESTONLYLIBS kokkos_gtest ${TEST_LINK_TARGETS}
 )
 
+TRIBITS_ADD_EXECUTABLE_AND_TEST(
+  UnitTest_PushFinalizeHook
+  SOURCES
+    UnitTest_PushFinalizeHook.cpp
```

Comment:
@mhoemmen I don't see this file with the tests in the PR, did it get left out?

Reply:
@ndellingwood @crtrott Argh!  Sorry about that; should I start a new PR to keep things atomic?

Reply:
No, I think adding fix commits to PRs has been fairly common

Reply:
@ibaned Cool; I'll direct the reviewers to squash rather than merge then.  Thanks!  Let me work on adding the file.

Reply:
@ndellingwood @ibaned @crtrott I just pushed a commit with the missing files.  Thanks!  Not sure why they didn't transfer over before.  (I started with a Trilinos patch, but the patch didn't merge, so I had to do it by hand.)